.github/workflows/ci.yml

```
@@ -0,0 +1,107 @@
+## (5b) Pin to commits, not tags
+
+name: CI Build and Check
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+env:
+  IMAGE: rancher/kube-api-auth
+
+permissions:
+  contents: read
+  security-events: write # upload Trivy Sarif results
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Get Sources
+      uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version-file: 'go.mod'
+        cache: false
+
+    - name: Validate the sources
+      run: |
+        ./scripts/validate
+        # Except for linting. That is done through the proper GH action
+
+    - name: Lint sources
+      uses: golangci/golangci-lint-action@v4
+
+    - name: Test
+      run: ./scripts/test
+
+  build:
+    runs-on: ubuntu-latest
+    needs:
+      - linter
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - linux/amd64
+          - linux/arm64
+    steps:
+    - name: Prepare Matrix Instance
+      run: |
+        platform=${{ matrix.platform }}
+        export GOOS="$(echo ${platform}   | sed -e 's|/.*||')"
+        export GOARCH="$(echo ${platform} | sed -e 's|.*/||')"
+        export ARCH="$GOARCH"
+        echo "GOOS=$GOOS"     >> $GITHUB_ENV
+        echo "GOARCH=$GOARCH" >> $GITHUB_ENV
+        echo "ARCH=$ARCH"     >> $GITHUB_ENV
```

You can use a [multi-dimensional matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-using-a-multi-dimension-matrix) in these cases:

```suggestion
      matrix:
        os: [ "linux" ]
        arch: [ "amd64", "arm64" ]
    steps:
    - name: Prepare Matrix Instance
      run: |
        export GOOS=${{ matrix.os }}
        export GOARCH=${{ matrix.arch }}
        export ARCH=${{ matrix.arch }}
        echo "GOOS=$GOOS"     >> $GITHUB_ENV
        echo "GOARCH=$GOARCH" >> $GITHUB_ENV
        echo "ARCH=$ARCH"     >> $GITHUB_ENV
```

Since we are using the `$GITHUB_ENV` I guess we can even skip the `export VAR`, and just have the last three lines. Not sure though, it will need a test.

---

Nice. Will do. The `export` was needed to make the variables in lines 54-56 usable in lines 57-59. With the simpler access to the parts I may  convert 54-56 to `echo` for proper log output.

---

Done with commit 301d15954ab974155391551b86e71f15bf5884c6